modules/react/popup/lib/Popper.tsx

```diff
@@ -140,6 +181,46 @@ const OpenPopper = React.forwardRef<HTMLDivElement, PopperProps>(
         },
       };
     }, [setPlacement, onPlacementChange]);
+ 
+    const fallbackPlacementsModifier = React.useMemo((): PopperJS.Modifier<any, any> => {
+      return {
+        name: 'fallbackPlacement',
+        enabled: true,
+        phase: 'main',
+        fn({state}){
+          const placements = [popperPlacement, getOppositePlacement(popperPlacement), ...baseFallbackPlacements]
+          const overflow = PopperJS.detectOverflow(state)
+          const preventOverflowData = state.modifiersData.preventOverflow
+          const isOverflowed = preventOverflowData && 
+          ((preventOverflowData.x === 0 && (overflow.right > 0 || overflow.left > 0)) || 
+          (preventOverflowData.y === 0 && (overflow.top > 0 || overflow.bottom > 0)))
+          maxRepositionCall.current --
+
+          for(let i = 0 ; i < placements.length; i++){
+            state.placement = placements[i]
+            const key = placements[i] as keyof PopperJS.SideObject
+            if(PopperJS.detectOverflow(state)[key] <= 0){
+              nextAvailablePlacementRef.current = placements[i]
+              break
+            }else{
+              state.placement = popperPlacement
+            }
+          } 
+
+          if(!isOverflowed && maxRepositionCall.current < 1){
+            // reset the max call if popper keeps open
+            maxRepositionCall.current = baseFallbackPlacements.length + 2
+          } else if(isOverflowed && maxRepositionCall.current < 1){
+            // when there is no available space on the four side, use the initial placement
+            state.reset = false
+          } else{           
+            // instruct Popper to run all modifier again
+            state.reset = true
+          }
+        },
+        requiresIfExists: ['preventOverflow']
```

Comment:
Add this to access to the preventOverflow data provided by preventOverflow modifier.